sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/CountIf.scala

```diff
@@ -34,8 +34,12 @@ import org.apache.spark.sql.types.{AbstractDataType, BooleanType}
   """,
   group = "agg_funcs",
   since = "3.0.0")
-case class CountIf(child: Expression) extends RuntimeReplaceableAggregate
-  with ImplicitCastInputTypes with UnaryLike[Expression] {
+case class CountIf(
+    child: Expression)
+  extends AggregateFunction
+  with ReplaceableAggregateFunction
```

Comment:
`extends AggregateFunction with ReplaceableAggregateFunction` looks a bit weird. Maybe the old name `RuntimeReplaceableAggregate` is better as a mix-in trait.